internal/providers/terraform/aws/elasticache_replication_group.go

```diff
@@ -3,27 +3,57 @@ package aws
 import (
 	"github.com/infracost/infracost/internal/resources/aws"
 	"github.com/infracost/infracost/internal/schema"
-	"github.com/tidwall/gjson"
 )
 
 func getElastiCacheReplicationGroupItem() *schema.RegistryItem {
 	return &schema.RegistryItem{
-		Name:  "aws_elasticache_replication_group",
-		RFunc: NewElastiCacheReplicationGroup,
+		Name:                "aws_elasticache_replication_group",
+		RFunc:               NewElastiCacheReplicationGroup,
+		ReferenceAttributes: []string{"aws_appautoscaling_target.resource_id"},
+		CustomRefIDFunc: func(d *schema.ResourceData) []string {
+			// returns a name that will match the custom format used by aws_appautoscaling_target.resource_id
+			name := d.Get("replication_group_id").String()
+			if name != "" {
+				return []string{"replication-group/" + name}
+			}
+			return nil
+		},
 	}
 }
 func NewElastiCacheReplicationGroup(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	cacheClusters := d.Get("num_cache_clusters").Int()
+	if d.IsEmpty("num_cache_clusters") {
+		// check for deprecated attribute
+		cacheClusters = d.Get("number_cache_clusters").Int()
+	}
+
+	clusterNodeGroups := d.Get("num_node_groups").Int()
+	if d.IsEmpty("num_node_groups") {
+		// check for deprecated attribute
+		clusterNodeGroups = d.Get("cluster_mode.0.num_node_groups").Int()
+	}
```

Comment:
```suggestion
	clusterNodeGroups := d. GetInt64OrDefault("num_node_groups", 1)
	if d.IsEmpty("num_node_groups") {
		// check for deprecated attribute
		clusterNodeGroups = d.Get("cluster_mode.0.num_node_groups").Int()
	}
```

same as above

Reply:
Hey just to double check, why do you think num_node_groups has a default of 1?  I'm not seeing that in the docs.

Reply:
yeah you're right - I must have seen misread the propery